lib/elixir/lib/system.ex

```diff
@@ -1103,9 +1112,18 @@ defmodule System do
   """
   @spec convert_time_unit(integer, time_unit | :native, time_unit | :native) :: integer
   def convert_time_unit(time, from_unit, to_unit) do
-    :erlang.convert_time_unit(time, normalize_time_unit(from_unit), normalize_time_unit(to_unit))
+    {from_conversion_ratio, from_erlang_unit} = normalize_time_unit(from_unit)
+    {to_conversion_ratio, to_erlang_unit} = normalize_time_unit(to_unit)
+
+    divide_conversion_ratio(
+      :erlang.convert_time_unit(time, from_erlang_unit, to_erlang_unit) * from_conversion_ratio,
+      to_conversion_ratio
+    )
   end
 
+  defp divide_conversion_ratio(time, 1), do: time
+  defp divide_conversion_ratio(time, conversion_ratio), do: trunc(time / conversion_ratio)
```

Comment:
Without this conditionally doing division, I was getting some test errors in `DateTime`. It appeared to be some rounding issues.